src/nupic/types/SdrProxy.hpp

```diff
@@ -55,52 +55,51 @@ class SDR_ReadOnly_ : public SDR
 };
 
 /**
- * SDR_Proxy class
+ * SDR_Reshape class
  *
  * ### Description
- * SDR_Proxy presents a view onto an SDR.
- *      + Proxies have the same value as their source SDR, at all times and
- *        automatically.
- *      + SDR_Proxy is a subclass of SDR and be safely typecast to an SDR.
- *      + Proxies can have different dimensions than their source SDR.
- *      + Proxies are read only.
+ * SDR_Reshape presents a view onto an SDR with different dimensions.
+ *      + SDR_Reshape is a subclass of SDR and be safely typecast to an SDR.
+ *      + The resulting SDR has the same value as the source SDR, at all times
+ *        and automatically.
+ *      + The resulting SDR is read only.
  *
- * SDR and SDR_Proxy classes tell each other when they are created and
- * destroyed.  Proxies can be created and destroyed as needed.  Proxies will
- * throw an exception if they are used after their source SDR has been
+ * SDR and SDR_Reshape classes tell each other when they are created and
+ * destroyed.  SDR_Reshape can be created and destroyed as needed.  SDR_Reshape
+ * will throw an exception if it is used after its source SDR has been
  * destroyed.
  *
  * Example Usage:
  *      // Convert SDR dimensions from (4 x 4) to (8 x 2)
- *      SDR       A(    { 4, 4 })
- *      SDR_Proxy B( A, { 8, 2 })
+ *      SDR         A(    { 4, 4 })
+ *      SDR_Reshape B( A, { 8, 2 })
  *      A.setCoordinates( {1, 1, 2}, {0, 1, 2}} )
  *      B.getCoordinates()  ->  {{2, 2, 5}, {0, 1, 0}}
  *
- * SDR_Proxy partially supports the Serializable interface.  SDR_Proxies can be
- * saved but can not be loaded.
+ * SDR_Reshape partially supports the Serializable interface.  SDR_Reshape can
+ * be saved but can not be loaded.
```

Comment:
put a note in this doc that we used to call this class SDR_Proxy, if someone runs into some older docs/conversations.

Reply:
Good point.  Instead of making a note of this I will do a proper search through the source code and find all references to words proxy & proxies.

Reply:
I found some references which I will review & fix:
```
$ grep -irni proxy src/
src/CMakeLists.txt:204:    nupic/types/SdrProxy.hpp
src/test/unit/types/SdrTest.cpp:20:#include <nupic/types/SdrProxy.hpp>
src/test/unit/types/SdrTest.cpp:748:    // Check proxy got all of the callbacks.
src/test/unit/types/SdrProxyTest.cpp:21:#include <nupic/types/SdrProxy.hpp>
src/test/unit/utils/SdrMetricsTest.cpp:22:#include <nupic/types/SdrProxy.hpp>
src/test/CMakeLists.txt:98:	   unit/types/SdrProxyTest.cpp
src/nupic/types/SdrProxy.hpp:22:#ifndef SDR_PROXY_HPP
src/nupic/types/SdrProxy.hpp:23:#define SDR_PROXY_HPP
src/nupic/types/SdrProxy.hpp:429:#endif // end ifndef SDR_PROXY_HPP
src/nupic/types/Sdr.hpp:199:     * before they're deallocated; SDR Proxy does this.

$ grep -irni proxies src/
src/test/unit/types/SdrTest.cpp:698:    // Test reshape proxies get callbacks
src/test/unit/types/SdrProxyTest.cpp:44:    // Test that proxies can be safely made and destroyed.
src/test/unit/types/SdrProxyTest.cpp:93:    // Cleanup remaining Proxies.
src/nupic/types/SdrProxy.hpp:135:            // getSparse() to get its data, and will use this proxies
```

```
$ grep -irni proxy bindings/
bindings/py/cpp_src/bindings/algorithms/py_SDR.cpp:27:#include <nupic/types/SdrProxy.hpp>
bindings/py/cpp_src/bindings/algorithms/py_SDR.cpp:304:SDR_Reshape supports pickle, however loading a pickled SDR proxy will return an

$ grep -irni proxies bindings/
no results.
```

Reply:
Done, there are no more references to `proxy` in the comments or docs.

Reply:
Thanks, I'd still be for a small note: 
"SDR_Reshaped used to be called SDR_Proxy"

Reply:
...as by the history I ment the gythub docs, PRs.

Reply:
Done, see latest commit.